### PR TITLE
Added Travis build script [WIP]

### PR DESCRIPTION
**Pull request by shouryaj**
Travis build script runs on all pushes and uploads artifacts to S3 bucket. 

Download mac distribution: http://travis-opendatakit-org.s3-website-us-west-2.amazonaws.com/{TRAVIS_BUILD_NUMBER}/dist/mac/ODK-XLSForm-Offline-macOS.zip

Download win distribution: http://travis-opendatakit-org.s3-website-us-west-2.amazonaws.com/{TRAVIS_BUILD_NUMBER}%5CODK%20XLSForm%20Offline.exe

Env variables:
ARTIFACTS_BUCKET
ARTIFACTS_KEY
ARTIFACTS_REGION
ARTIFACTS_SECRET

Notes: 
1. The links above don't work right now because access is still not public.
2. The ./make-dist.sh script doesn't run on windows so I only provide the Mac zipped version.
3. Path to store in S3 is somewhat flexible so if something else is preferred let me know.
4. Right now, README says all of this needs Python 2.7, so build script just requires 2.7. If this needs to be updated to some version of Python 3.*, should be an easy add.
5. I'm currently using my own travis account, but if everything looks okay, I can create a travis account for ODK and move everything over.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+python:
+  - "2.7"
+
+addons:
+  artifacts:
+    paths:
+      - ./dist/mac/ODK-XLSForm-Offline-macOS.zip
+      - ./dist/win
+    target_paths:
+      - ${TRAVIS_REPO_SLUG}_${TRAVIS_BUILD_NUMBER}_${TRAVIS_JOB_NUMBER}
```
Comment (yanokwa) [inline .travis.yml]: I think we can drop this for now and leave the default.

File: README.md [hunk 2]
```diff
@@ -2,6 +2,7 @@
 ![Platform](https://img.shields.io/badge/platform-Python-blue.svg)
 [![License](https://img.shields.io/badge/license-Apache%202.0-blue.svg)](https://opensource.org/licenses/Apache-2.0)
 [![Slack status](http://slack.opendatakit.org/badge.svg)](http://slack.opendatakit.org)
+![Build status](https://travis-ci.com/shouryaj/xlsform-offline.svg?branch=master)
```
Comment (yanokwa) [inline README.md]: Replace shouryaj with opendatakit

File: make-dist.sh [hunk 3]
```diff
@@ -1,29 +1,29 @@
 #!/bin/bash
 set -e
 
-VERSION=$(grep VERSION src/main.py | sed -e "s|.*'\(.*\)'|\1|")
```
Comment (yanokwa) [inline make-dist.sh]: I'm curious why version was dropped.

Comment (shouryaj) [reply make-dist.sh]: I was having some trouble with getting travis to use escape characters in its path to upload the correct file to S3. Is this an issue?

**Comment by shouryaj** [comment]
> Please update the PR description to say what specific env variables are needed. Once I have that, I can merge to a long-running branch on the repo so we can see things actually working on Travis.

Done!

> If you can rebase and squash your commits so the history is clean and tells a story. Yes I can do that on merge, but it makes review easier to have that clean history.

Sounds good, I'll work on squashing the commits!

File: make-dist.sh (hunk 3 above)
Comment (yanokwa) [reply make-dist.sh]: It's nice to have some way to know version we are working with. One better approach than the static version we have is that if the build machine runs git, you could have it use the hash (`git describe --tags --dirty --always`) This is what we do with a lot of the other ODK tools.

https://github.com/opendatakit/briefcase/blob/master/build.gradle#L99

**Comment by yanokwa** [comment]
Squashing and merging to a long running travis branch to test the build.

**Pull request merged**